tests/test_relative_free_energy.py

```diff
@@ -183,6 +179,55 @@ def test_steps_per_frames():
         np.testing.assert_array_equal(frame[-1], test_frame[-1])
 
 
+def test_imaging_frames():
+    """Should not be able to run a relative free energy calculation with a single window"""
+    mol_a, mol_b, core = get_hif2a_ligand_pair_single_topology()
+    forcefield = Forcefield.load_from_file(DEFAULT_FF)
+    seed = 2022
+    frames = 1
+    steps_per_frame = 100
+    equil_steps = 1
+    windows = 2
+    res, _ = run_solvent(
+        mol_a,
+        mol_b,
+        core,
+        forcefield,
+        None,
+        frames,
+        seed,
+        n_eq_steps=equil_steps,
+        steps_per_frame=steps_per_frame,
+        n_windows=windows,
+        image_traj=False,
+    )
+    keep_idxs = [0, len(res.initial_states) - 1]
+    assert len(keep_idxs) == len(res.frames)
+
+    # A buffer, as imaging doesn't ensure everything is perfectly in the box
+    padding = 0.3
+    for i in range(len(res.frames)):
+        initial_state = res.initial_states[keep_idxs[i]]
+        box_center = compute_box_center(res.boxes[i][0])
+        box_extents = np.max(res.boxes[i], axis=(0, 1))
+
+        # Verify that coordinates are either outside of the box or below zero
+        assert np.any(np.abs(np.max(res.frames[i], axis=(0, 1))) > box_extents + padding) or np.any(
+            np.min(res.frames[i], axis=(0, 1)) < -padding
+        )
+        # Ligand won't be near center of box
+        assert not np.allclose(np.mean(res.frames[i][0][initial_state.ligand_idxs], axis=0), box_center)
+
+        imaged = image_frames(initial_state, res.frames[i], res.boxes[i])
+
+        # Verify that after imaged, coordinates are within padding of the box extents
+        assert np.all(np.abs(np.max(imaged, axis=(0, 1))) <= box_extents + padding) and np.any(
```

Comment:
Should the second condition also be `np.all`?

Reply:
Both fixed in [3b58a0b](https://github.com/proteneer/timemachine/pull/905/commits/3b58a0ba366ae6c2d9a7da2446961aa52271ccee)